### ISSUE TITLE
ReceiverDisconnectedException

### ISSUE THREAD

**Issue by amybachir** [Azure/azure-event-hubs-spark#545]
I saw multiple issues open around ReceiverDisconnectedException and all the answers point to having multiple Partitions Receiver but that is not the case in my situation. So, what is this exception all about anyway?

I'm not  running two instances of the application, or two concurrent classes, or two applications that use the same event hub consumer group.

I'm running the exact same logic in this link https://github.com/Azure/AI-PredictiveMaintenance/blob/master/src/SparkJobs/Featurization/src/main/scala/com/microsoft/ciqs/predictivemaintenance/CycleAggregator.scala

Here are my SBT dependencies:
```
scalaVersion := "2.11.12"

libraryDependencies ++= Seq(
  "org.apache.spark" %% "spark-core" % "2.3.0",
  "org.apache.spark" %% "spark-streaming" % "2.3.0",
  "org.apache.spark" %% "spark-sql" % "2.3.0",
  "com.microsoft.azure" %% "azure-eventhubs-spark" % "2.3.17",
  "com.microsoft.azure" % "azure-storage" % "7.0.0",
  "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.11.2",
  "com.fasterxml.jackson.core" % "jackson-databind" % "2.11.2",
  "com.fasterxml.jackson.core" % "jackson-core" % "2.11.2"
)

libraryDependencies += "org.scalatest" %% "scalatest" % "3.0.5" % "test"

```

And here's the error message:

```
[error] Caused by: com.microsoft.azure.eventhubs.ReceiverDisconnectedException: New receiver 'spark-driver-1205' with higher epoch of '0' is created hence current receiver 'spark-driver-1204' with epoch '0' is getting disconnected. If you are recreating the receiver, make sure a higher epoch is used. TrackingId:aae23a6c0000753a000b49f85f8667a3_G19_B41, SystemTracker:iot-predictive-maintenance-eventhub:eventhub:iot-telemetry-stream~32766|$default, Timestamp:2020-10-14T02:51:20, errorContext[NS: iot-predictive-maintenance-eventhub.servicebus.windows.net, PATH: iot-telemetry-stream/ConsumerGroups/$Default/Partitions/0, REFERENCE_ID: LN_508d8f_1602643872371_af3_G19, PREFETCH_COUNT: 500, LINK_CREDIT: 500, PREFETCH_Q_LEN: 0]
[error]         at com.microsoft.azure.eventhubs.impl.ExceptionUtil.toException(ExceptionUtil.java:43)
[error]         at com.microsoft.azure.eventhubs.impl.MessageReceiver.onClose(MessageReceiver.java:789)
[error]         at com.microsoft.azure.eventhubs.impl.BaseLinkHandler.processOnClose(BaseLinkHandler.java:73)
[error]         at com.microsoft.azure.eventhubs.impl.BaseLinkHandler.handleRemoteLinkClosed(BaseLinkHandler.java:109)
[error]         at com.microsoft.azure.eventhubs.impl.BaseLinkHandler.onLinkRemoteClose(BaseLinkHandler.java:51)
[error]         at org.apache.qpid.proton.engine.BaseHandler.handle(BaseHandler.java:176)
[error]         at org.apache.qpid.proton.engine.impl.EventImpl.dispatch(EventImpl.java:108)
[error]         at org.apache.qpid.proton.reactor.impl.ReactorImpl.dispatch(ReactorImpl.java:324)
[error]         at org.apache.qpid.proton.reactor.impl.ReactorImpl.process(ReactorImpl.java:291)
[error]         at com.microsoft.azure.eventhubs.impl.MessagingFactory$RunReactor.run(MessagingFactory.java:766)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
```

**Comment by nyaghma** [comment]
@amybachir The error indicates that multiple receivers are being opened to the same partition-consumer group combo at the same time. This could happen even within a single application if the application runs several actions against a stream or writes a stream to multiple sinks without caching it, which results in recomputing the stream. Please refer to the [multiple-readers-example](https://github.com/Azure/azure-event-hubs-spark/blob/master/examples/multiple-readers-example.md) document for more information.

**Comment by amybachir** [comment]
@nyaghma  thanks for the quick response! 

So this weird! I'm no longer seeing the error but the solution was very unexpected. I set the Azure event hub partition to 1 by a mistake but the Azure documentation says it must be between 2 and 32. Once I changed the event hub partition, I was able to stream data. I don't know why Azure even allowed me to go with a default of 1 partition!

![image](https://user-images.githubusercontent.com/19543576/96275558-e02a0900-0f9f-11eb-9c65-532ffeeda1b1.png)
